plane/src/drone/docker/commands.rs

```diff
@@ -187,3 +195,139 @@ pub async fn run_container(
+        let _container_id = ContainerId::from(format!("plane-test-{}", backend_name));
+        let mut executor_config = ExecutorConfig::from_image_with_defaults("debian:bookworm");
+
+        let resource_limits: ResourceLimits = serde_json::from_str(
+            r#"
+        {
+        "cpu_period": 1000000,
+        "cpu_time_limit": 100,
+        "cpu_period_percent": 80,
+        "memory_limit_bytes": 10000000,
+        "disk_limit_bytes": 10000000000
+        }
+        "#,
+        )?;
+
+        executor_config.resource_limits = resource_limits;
+        let info = docker.info().await?;
+        match info.driver {
+            Some(t) if matches!(t.as_str(), "btrfs" | "zfs") => {}
+            Some(t)
+                if t == "overlay2" && {
+                    let fs_ok;
+                    if let Some(status) = info.driver_status {
+                        fs_ok = !status.iter().flatten().any(|s| s.contains("xfs"));
+                    } else {
+                        fs_ok = false
+                    };
+                    !fs_ok
+                } =>
+            {
+                //disk limits not supported with xfs backend for overlay 2
+                executor_config.resource_limits.disk_limit_bytes = None;
+            }
+            _ => {
+                //disk limits not supported otherwise
+                executor_config.resource_limits.disk_limit_bytes = None;
+            }
+        }
+
+        let mut config =
+            get_container_config_from_executor_config(&backend_name, executor_config, &None)?;
+        config.cmd = Some(vec!["echo".into(), "hello world".into()]);
+        let out = run_container_with_config(config.clone()).await?;
+        assert_eq!(out.trim(), "hello world".to_string());
+
+        config.cmd = None;
+        config.entrypoint = Some(vec!["sh".into(), "-c".into(), "ulimit -t".into()]);
+        let out = run_container_with_config(config.clone()).await?;
+        assert_eq!(out.trim(), "100".to_string());
+
+        config.entrypoint = None;
+        config.cmd = Some(vec!["cat".into(), "/sys/fs/cgroup/cpu.max".into()]);
+        let out = run_container_with_config(config.clone()).await?;
+        assert_eq!(out.trim(), "800000 1000000".to_string());
+
+        //NOTE: memory limit rounds down to nearest page size multiple,
+        // this assumes the page size to be 4096 bytes.
+        let mem_limit_corr = 10000000 - (10000000 % 4096);
+        config.cmd = Some(vec!["cat".into(), "/sys/fs/cgroup/memory.max".into()]);
+        let out = run_container_with_config(config.clone()).await?;
+        assert_eq!(out.trim(), mem_limit_corr.to_string());
+
+        //unfortunately, the docker disk limit is dependent on the storage backend
+        //hence just validating here that config is as expected.
+        config.host_config.unwrap().storage_opt.map(|hm| {
+            let size = hm.get("size").cloned();
+            assert_eq!(size, Some("10000000000".to_string()));
+        });
```

Comment:
If I'm reading this right, it doesn't catch the case where `get_container_config_from_executor_config` should set `storage_opt` to a `Some(_)` but doesn't.

Instead, consider:

```rust
if let Some(expected_size) = executor_config.resource_limits.disk_limit_bytes {
    assert_eq!(
        expected_size.to_string(),
        config.host_config
            .unwrap()
            .storage_opt
            .unwrap()
            .get("size")
            .unwrap()
    );
}
```